packages/server/tests/acceptance/estimateGasPrecompile.spec.ts

```diff
@@ -0,0 +1,473 @@
+      contractMirror.contract_id,
+      200,
+      relay.provider,
+      requestId,
+    );
+    accounts[1] = await servicesNode.createAccountWithContractIdKey(
+      contractMirror.contract_id,
+      30,
+      relay.provider,
+      requestId,
+    );
+    accounts[2] = await servicesNode.createAccountWithContractIdKey(
+      contractMirror.contract_id,
+      30,
+      relay.provider,
+      requestId,
+    );
+
+    contract = new ethers.Contract(
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[0].wallet,
+    );
+
+    // Create token and NFT contracts
+    tokenAddress = await createFungibleToken();
+    nftAddress = await createNft();
+
+    let estimateContractSigner1 = new ethers.Contract(
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[1].wallet,
+    );
+    const tx1_ft = await estimateContractSigner1.associateTokenExternal(
+      accounts[1].wallet.address,
+      tokenAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    let estimateContractSigner2 = new ethers.Contract(
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[2].wallet,
+    );
+    const tx2_ft = await estimateContractSigner2.associateTokenExternal(
+      accounts[2].wallet.address,
+      tokenAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    const t1_nft = await estimateContractSigner1.associateTokenExternal(
+      accounts[1].wallet.address,
+      nftAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    const tx2_nft = await estimateContractSigner2.associateTokenExternal(
+      accounts[2].wallet.address,
+      nftAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    await new Promise((r) => setTimeout(r, 2500));
+    //NFT
+    await contract.grantTokenKycExternal(nftAddress, accounts[0].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[1].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[2].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, prefix + EstimatePrecompileContractAddress, { gasLimit: 500_000 });
+
+    //Fungible
+    await contract.grantTokenKycExternal(tokenAddress, accounts[0].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[1].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[2].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, prefix + EstimatePrecompileContractAddress, {
+      gasLimit: 500_000,
+    });
+  });
+
+  const baseGasCheck = (estimatedGasValue, expectedValue: number) => {
+    // handle deviation of 20%
+    expect(Number(estimatedGasValue)).to.be.lessThan(expectedValue * 1.4);
+  };
```

Comment:
Currently this checks if estimateGas value is less than 140% of the expectedValue. This should be tweaked to accept lower bound and upper bound params to set a certain deviation range, like 5%-20%. This way, we can precisely control how much `estimatedGasValue` can differ from `expectedValue`.